src/algebra/pi_instances.lean

```diff
@@ -240,15 +240,25 @@ lemma snd.is_monoid_hom [monoid α] [monoid β] : is_monoid_hom (prod.snd : α 
 { map_mul := λ _ _, rfl, map_one := rfl }
 
 /-- Given monoids `α, β`, the natural projection homomorphism from `α × β` to `α`. -/
-@[to_additive prod.add_monoid_hom.fst "Given add_monoids `α, β`, the natural projection homomorphism from `α × β` to `α`."]
+@[to_additive prod.add_monoid_hom.fst "Given `add_monoid`s `α, β`, the natural projection homomorphism from `α × β` to `α`."]
 def monoid_hom.fst [monoid α] [monoid β] : α × β →* α :=
 ⟨λ x, x.1, rfl, λ _ _, prod.fst_mul⟩
 
 /-- Given monoids `α, β`, the natural projection homomorphism from `α × β` to `β`.-/
-@[to_additive prod.add_monoid_hom.snd "Given add_monoids `α, β`, the natural projection homomorphism from `α × β` to `β`."]
+@[to_additive prod.add_monoid_hom.snd "Given `add_monoid`s `α, β`, the natural projection homomorphism from `α × β` to `β`."]
 def monoid_hom.snd [monoid α] [monoid β] : α × β →* β :=
 ⟨λ x, x.2, rfl, λ _ _, prod.snd_mul⟩
 
+/-- Given monoids `α, β`, the natural inclusion homomorphism from `α` to `α × β`. -/
+@[to_additive prod.add_monoid_hom.inl "Given `add_monoid`s `α, β`, the natural inclusion homomorphism from `α` to `α × β`."]
+def monoid_hom.inl [monoid α] [monoid β] : α →* α × β :=
+⟨λ x, (x, 1), rfl, λ x y, show _ = (_, _), by rw mul_one⟩
```

Comment:
For the additive version we also have unbundled `prod.inl` / `prod.inr`. Please mention this fact in the docstrings.